ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/BlockSimulator.java

```diff
@@ -340,7 +346,7 @@ private BlockSimulationResult createFinalBlock(
             .transactionsRoot(BodyValidation.transactionsRoot(transactions))
             .receiptsRoot(BodyValidation.receiptsRoot(receipts))
             .logsBloom(BodyValidation.logsBloom(receipts))
-            .gasUsed(blockStateCallSimulationResult.getCumulativeGasUsed())
+            .gasUsed(simResult.getCumulativeGasUsed())
```

Comment:
they said they will need prevRandao, it is possible to do it like that ? It is similar to engineNewPayload

Reply:
oh yes got it